library/bignum_mod_raw.c

```diff
@@ -119,7 +119,16 @@ int mbedtls_mpi_mod_raw_write( const mbedtls_mpi_uint *A,
 /* END MERGE SLOT 4 */
 
 /* BEGIN MERGE SLOT 5 */
-
+void mbedtls_mpi_mod_raw_add( mbedtls_mpi_uint *X,
+                              const mbedtls_mpi_uint *A,
+                              const mbedtls_mpi_uint *B,
+                              const mbedtls_mpi_mod_modulus *N )
+{
+    mbedtls_mpi_uint carry, borrow;
+    carry  = mbedtls_mpi_core_add( X, A, B, N->limbs );
+    borrow = mbedtls_mpi_core_sub( X, X, N->p, N->limbs );
```

Comment:
From the documentation on core_add_if

~~~~~~
 * \warning          If \p cond is neither 0 nor 1, the result of this function
 *                   is unspecified, and the resulting value in \p X might be
 *                   neither its original value nor \p X + \p A.
~~~~~~

I appreciate that mod_raw lies in a grey zone between the high level and the core api's but should we make sure that we passing a valid input that satisfies this requirement -> `( carry ^ borrow ) == 0 or ( carry ^ borrow )  == 1` ?

Update looking at the code it could also work for  `( carry ^ borrow ) >= 0` ?

Reply:
`carry` and `borrow` can both only be `0` or `1`, so `carry ^ borrow` will be only `0` or `1`